src/libstore/build.cc

```diff
@@ -2499,13 +2503,20 @@ void DerivationGoal::runChild()
             }
             sandboxProfile += ")\n";
 
+            sandboxProfile += additionalSandboxProfile;
+
             debug("Generated sandbox profile:");
             debug(sandboxProfile);
 
+            char *tmpProfile = strdup((format("%1%/nix-sandboxXXXXXX.sb") % globalTmpDir).str().c_str());
+            int profileFd = mkstemps(tmpProfile, 3);
+            closeOnExec(profileFd);
+            writeFull(profileFd, sandboxProfile);
```

Comment:
Still not following. The temp profile here contains allow-file-read for all of the derivation's inputs as well as its custom profile. That list of inputs isn't known by the derivation itself.